src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java

```diff
@@ -0,0 +1,220 @@
+    extends AbstractCheck {
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY = "ws.preceded";
+
+    @Override
+    public int[] getDefaultTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getRequiredTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return new int[] {TokenTypes.COLON};
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        if (isInSwitch(ast) && isWhiteSpaceBeforeColon(ast)) {
+            log(ast, MSG_KEY, ast.getText());
+        }
+    }
+
+    /**
+     * Checks if the colon is inside a switch block.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if colon case is inside a switch block.
+     */
+    private static boolean isInSwitch(DetailAST colonAst) {
+        return TokenUtil.isOfType(colonAst.getParent(), TokenTypes.LITERAL_CASE,
+                TokenTypes.LITERAL_DEFAULT);
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case or switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhiteSpaceBeforeColon(DetailAST colonAst) {
+        final DetailAST parent = colonAst.getParent();
+        final boolean result;
+        if (isOnDifferentLineWithPreviousToken(colonAst)) {
+            result = true;
+        }
+        else if (parent.getType() == TokenTypes.LITERAL_CASE) {
+            result = isWhitespaceBeforeColonOfCase(colonAst);
+        }
+        else {
+            result = isWhitespaceBeforeColonOfDefault(colonAst);
+        }
+        return result;
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch case.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhitespaceBeforeColonOfCase(DetailAST colonAst) {
+        final DetailAST previousSibling = colonAst.getPreviousSibling();
+        int offset = 0;
+        if (previousSibling.getType() == TokenTypes.BLOCK_COMMENT_BEGIN) {
+            offset = 1;
+        }
+        return colonAst.getColumnNo() != getLastColumnNumberOf(previousSibling) + offset;
+    }
+
+    /**
+     * Checks if there is a whitespace before the colon of a switch default.
+     *
+     * @param colonAst DetailAST to check.
+     * @return true, if there is whitespace preceding colonAst.
+     */
+    private static boolean isWhitespaceBeforeColonOfDefault(DetailAST colonAst) {
+        final boolean result;
+        final DetailAST previousSibling = colonAst.getPreviousSibling();
+        if (previousSibling == null) {
```

Comment:
why in default processing you do not have code like
`if (previousSibling.getType() == TokenTypes.BLOCK_COMMENT_BEGIN) {` ?

Reply:
Please see https://github.com/checkstyle/checkstyle/pull/9728#discussion_r633028187

Reply:
yes it is possible to have comments.
But why it is different between `case:` and `default:` ?
they are case by structure 

`colonAst.getColumnNo() != getLastColumnNumberOf(previousSibling) + offset;`
vs
`colonAst.getColumnNo() != getLastColumnNumberOf(previousSibling) + 1;`

Reply:
There is a difference because for `LITERAL_CASE` we always have an `EXPR` before the colon, for which the `offset` is 0.
When there is `BLOCK_COMMENT`, we make offset to be 1 otherwise it is 0 as shown in the below code.

https://github.com/checkstyle/checkstyle/blob/bfe149624e6a9003c2f7bef163c4c8bea39bca7b/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceBeforeCaseDefaultColonCheck.java#L160-L164

Now, for `LITERAL_DEFAULT` we are sure that there can only be `BLOCK_COMMENT`, nothing else can be used there. So we are directly adding 1, whenever there is a previous sibling. 

Hence, we are treating `BLOCK_COMMENT` same in both cases i.e. we add an `offset` of 1 in both cases.

Reply:
ok now I see it
```
|--CASE_GROUP -> CASE_GROUP [5:12]
|   |--LITERAL_CASE -> case [5:12]
|   |   |--EXPR -> EXPR [5:17]
|   |   |   `--NUM_INT -> 1 [5:17]
|   |   `--COLON -> : [5:18]
|   `--SLIST -> SLIST [6:16]
|       `--LITERAL_BREAK -> break [6:16]
|           `--SEMI -> ; [6:21]
|--CASE_GROUP -> CASE_GROUP [7:12]
|   |--LITERAL_DEFAULT -> default [7:12]
|   |   `--COLON -> : [7:20]
```